lib/hex_web/store/s3.ex

```diff
@@ -90,33 +87,15 @@ defmodule HexWeb.Store.S3 do
     HexWeb.Plug.redirect(conn, url)
   end
 
-  def upload(bucket, path, headers, data) do
+  def upload(bucket, path, data, opts \\ []) do
```

Comment:
Putting opts last and making them optional really simplifies a lot of the calls to it in this file.